### ISSUE TITLE
SX1278 cann't receive packets from USRP correctly with SF12

### ISSUE THREAD

**Issue by JaysonJie1111** [tapparelj/gr-lora_sdr#33]
Hi, This is JaysonJie. I discuss the issue when I used the SX1278 to send a packet to usrp with SF 12, but I only received  a set of wrong characters。As shown in the figure below:
![2022-11-03 16-49-30 的屏幕截图](https://user-images.githubusercontent.com/117350017/199942764-8a00167c-fda7-450f-9b05-83b0dcb89141.png)
But, when I try to use other SFs (like 7,8,9,10,11)  to send messages through SX1278, usrp can receive the correct message。As shown in the figure below:
![2022-11-04 16-16-18 的屏幕截图](https://user-images.githubusercontent.com/117350017/199943482-c605ba69-7d89-4cb5-8e86-17fb23157829.png)
So, how can i deal with this problem?

(By the way, i also try to use usrp to send the message to SX1278 with SF 12, It didn't work. But in other SF cases, it worked! )

**Comment by tapparelj** [comment]
Hi @JaysonJie1111,
I suspect that it is related to a mismatch of the setting of low datarate optimisation (LDRO). 
What bandwidth are you using?
By the way is it normal that you send two special characters in the payload beginning?

**Comment by JaysonJie1111** [comment]
I have used 433.0MHz and 470.0MHz  in bandwidth.
These two special characters is the rf95 header in the payload beginning of SX1278.

**Comment by tapparelj** [comment]
- Are you speaking of the carrier frequency ? because LoRa  usually uses a bandwidth between 125kHz and 500kHz.
- I see for the symbols. There is a block prepending the radiohead header to the payload that you can use when transmitting. You can insert it just before the whitening block (the destination can be set to 255 for broadcast).

**Comment by JaysonJie1111** [comment]
- Sorry, I confused the bandwidth  with the  carrier frequency. I used 250kHz in bandwidth. And the central carrier frequency is 433.0MHz and 470.0MHz.
- Yes, when I use USRP to send packets to SX1278, I use this block just like you said. It made the usrp send the packet successfully in SF 7,8,9,10,11. but it also didn't work in SF12.
-

**Comment by tapparelj** [comment]
LDRO should be enabled when a symbol duration exceed 16ms. Since you use a bandwidth of 250kHz and sf12, you have a symbol duration of $\frac{2^{12}}{250}=16.384ms$. 
Some libraries don't respect the 16ms criterion so you can either:
-  Enable it yourself by setting the register RegModemConfig3 (0x26) (see [datasheets](https://cdn-shop.adafruit.com/product-files/3179/sx1276_77_78_79.pdf) page 114 ).
- Disable it by changing the parameter LDRO in the interleaver block to "Disable"

**Comment by JaysonJie1111** [comment]
Thanks! I found that we were using an older version of the tapparelj/gr-lora_sdr which is without low datarate optimisation support.
![2022-11-06 17-00-25 的屏幕截图](https://user-images.githubusercontent.com/117350017/200177082-ddba3715-4f0a-4ef3-913f-9be30005b60e.png)
And then we tried to disable the LDRO function on our COTS node.
![2](https://user-images.githubusercontent.com/117350017/200177254-a7f928b6-2c92-46bf-abc5-3826637b0fd4.png)
Finally, it work!!!  The USRP can receive the information from the COTS node correctly under the SF = 12 condition.
The COTS node also can receive the information from the USRP correctly under the SF = 12 condition.

We will update the **tapparelj/gr-lora_sdr** we used to the latest version to avoid this problem again.
